resources/assets/js/components/JobTasks/JobTasks.tsx

```diff
@@ -1,222 +1,387 @@
+                                          defaultMessage="You have exceeded the maximum number of key tasks allowed, but that’s okay. You can continue to add key tasks as you brainstorm here, but you will be asked to trim your list to 6 key tasks or fewer to proceed."
+                                        />
+                                      </p>
+                                    </div>
+                                  </div>
+                                </div>
+                              )}
+                              <div
+                                key={`listItem${index}`}
+                                className={`job-builder-task${
+                                  index + 1 > validCount! ? " invalid" : ""
+                                }`}
+                                data-c-grid-item="base(1of1)"
+                                data-c-input="textarea"
+                                data-tc-up-down-item
+                              >
+                                <div>
+                                  <button
+                                    type="button"
+                                    data-tc-move-up-trigger
+                                    onClick={(): void =>
+                                      arrayHelpers.move(index, index - 1)
+                                    }
+                                  >
+                                    <i className="fas fa-long-arrow-alt-up" />
+                                  </button>
+                                  <button
+                                    type="button"
+                                    data-tc-move-down-trigger
+                                    onClick={(): void =>
+                                      arrayHelpers.move(index, index + 1)
+                                    }
+                                  >
+                                    <i className="fas fa-long-arrow-alt-down" />
+                                  </button>
+                                </div>
+                                <div>
+                                  <button
+                                    type="button"
+                                    data-tc-builder-task-delete-trigger
+                                    onClick={(): void => {
+                                      arrayHelpers.remove(index);
+                                    }}
+                                  >
+                                    <i
+                                      className="fas fa-trash"
+                                      data-c-colour="stop"
+                                    />
+                                  </button>
+                                </div>
+                                <Field
+                                  id={`task${task.id}`}
+                                  name={`tasks.${index}.description`}
```

Comment:
I would have expected using index like this not to work, but it tests fine in storybook. I guess arrayHelpers makes it work?

Reply:
It's a combo of Formik allowing it and having a helper function to grab the nested object (if it exists). I had to update the TextAreaInput component to make use of it, but it seems to work fine for array nested inputs as well as individual ones.

https://jaredpalmer.com/formik/docs/api/fieldarray#fieldarray-array-of-objects